src/ragger/conftest/base_conftest.py

```diff
@@ -170,6 +171,17 @@ def create_backend(root_pytest_dir: Path, backend_name: str, firmware: Firmware,
 def backend(root_pytest_dir, backend_name, firmware, display, log_apdu_file, cli_user_seed):
     with create_backend(root_pytest_dir, backend_name, firmware, display, log_apdu_file,
                         cli_user_seed) as b:
+        if backend_name.lower() == "ledgerwallet" and conf.OPTIONAL.APP_NAME:
+            # Make sure the app is restarted as this is what is requested by the fixture scope
+            app_name, version = get_current_app_name_and_version(b)
+            requested_app = conf.OPTIONAL.APP_NAME
+            if app_name == requested_app:
+                exit_current_app(b)
+                b.handle_usb_reset()
```

Comment:
As discussed, this should be kept, App that don't need it should use a better scope than function scope to avoid having to many app consent to give.
And signing the app is still a possibility!